cns/service/main.go

```diff
@@ -651,16 +651,14 @@ func main() {
 		return
 	}
 
+	// copy ChannelMode from cnsconfig to HTTPRemoteRestService config
+	config.ChannelMode = cnsconfig.ChannelMode
 	if cnsconfig.ChannelMode == cns.Managed {
-		config.ChannelMode = cns.Managed
 		privateEndpoint = cnsconfig.ManagedSettings.PrivateEndpoint
 		infravnet = cnsconfig.ManagedSettings.InfrastructureNetworkID
 		nodeID = cnsconfig.ManagedSettings.NodeID
-	} else if cnsconfig.ChannelMode == cns.CRD {
-		config.ChannelMode = cns.CRD
-	} else if cnsconfig.ChannelMode == cns.MultiTenantCRD {
-		config.ChannelMode = cns.MultiTenantCRD
-	} else if acn.GetArg(acn.OptManaged).(bool) {
+	}
+	if isManaged, ok := acn.GetArg(acn.OptManaged).(bool); ok && isManaged {
```

Comment:
i think to fully match the prior behavior, this should only happen if `cnsconfig.ChannelMode` is unset (`strings.IsEmpty(cnsconfig.ChannelMode)`) AND the `OptManaged` flag is set

Reply:
`cnsconfig.ChannelMode` can't be empty because we set it to `Direct` by default: https://github.com/jackieluc/azure-container-networking/blob/50042744c6edbfa50e82826fb540cad979226b5c/cns/configuration/configuration.go#L207-L209